src/plugins/explore/public/application/utils/state_management/actions/processors/trace_aggregation_processor.ts

```diff
@@ -0,0 +1,471 @@
+}
+
+export function extractPPLIntervalMs(results: ISearchResult, fallbackMs: number): number {
+  if (!results.hits?.hits?.[0]?._source) {
+    return fallbackMs;
+  }
+
+  const firstHit = results.hits.hits[0];
+  const spanTimeKey = Object.keys(firstHit._source).find((key) => key.startsWith('span('));
+
+  if (!spanTimeKey) {
+    return fallbackMs;
+  }
+
+  const intervalMatch = spanTimeKey.match(/span\([^,]+,\s*(\d+)([smhd])\)/);
+  if (!intervalMatch) {
+    return fallbackMs;
+  }
+
+  const [, value, unit] = intervalMatch;
+  const num = parseInt(value, 10);
+
+  switch (unit) {
+    case 's':
+      return num * 1000;
+    case 'm':
+      return num * 60 * 1000;
+    case 'h':
+      return num * 60 * 60 * 1000;
+    case 'd':
+      return num * 24 * 60 * 60 * 1000;
+    default:
+      return fallbackMs;
+  }
+}
+
+function processRequestCountData(
+  results: ISearchResult,
+  dataset: DataView,
+  minTime: number,
+  maxTime: number,
+  intervalMs: number,
+  timeField: string,
+  xAxisFormat: { id: string; params: { pattern: string } }
+): ChartData {
+  const chartValues: Array<{ x: number; y: number }> = [];
+  const xAxisOrderedValues: number[] = [];
+  let actualIntervalMs = intervalMs;
+
+  if (results.hits?.hits) {
+    const dataMap = new Map<number, number>();
+    let totalCount = 0;
+
+    actualIntervalMs = extractPPLIntervalMs(results, intervalMs);
+
+    if (!Number.isFinite(actualIntervalMs) || actualIntervalMs <= 0) {
+      return {
+        values: chartValues,
+        xAxisOrderedValues,
+        xAxisFormat,
+        xAxisLabel: timeField,
+        yAxisLabel: 'Request Count',
+        ordered: {
+          date: true,
+          interval: moment.duration(intervalMs),
+          intervalOpenSearchUnit: 'ms',
+          intervalOpenSearchValue: intervalMs,
+          min: moment(minTime),
+          max: moment(maxTime),
+        },
+      };
+    }
+
+    for (const hit of results.hits.hits) {
+      // Look for span() field or direct time field
+      const spanTimeKey = Object.keys(hit._source || {}).find((key) => key.startsWith('span('));
+      const timeValue = spanTimeKey ? hit._source?.[spanTimeKey] : hit._source?.[timeField];
+
+      const requestCount = hit._source?.['request_count'] || hit._source?.['count()'] || 0;
+
+      totalCount += requestCount;
+
+      if (timeValue && !isNaN(requestCount)) {
+        const timestamp = new Date(timeValue).getTime();
+        if (!isNaN(timestamp)) {
+          const bucketKey = Math.floor(timestamp / actualIntervalMs) * actualIntervalMs;
+          dataMap.set(bucketKey, (dataMap.get(bucketKey) || 0) + requestCount);
+        }
+      }
+    }
+
+    const startBucket = Math.floor(minTime / actualIntervalMs) * actualIntervalMs;
+    const endBucket = Math.floor(maxTime / actualIntervalMs) * actualIntervalMs;
+
+    for (let bucketKey = startBucket; bucketKey <= endBucket; bucketKey += actualIntervalMs) {
+      const value = dataMap.get(bucketKey) || 0;
```

Comment:
Is there a possibility that the time range is large and interval is small, this could generate excessive buckets. Should we have a maxBuckets size?

Reply:
Currently the interval selection behavior is disabled. It defaults to auto which has the safeguards for excessive buckets. Once the reported bug is fixed if we decide to re-enable user manual interval selection I will further test this and verify no issues.